unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -397,64 +398,90 @@ private async UniTask<WearableItem> SyncWearablesRequestsAsync(string newWearabl
+                    WearableItem wearable = new WearableItem
                     {
-                        foreach (var representationContent in representation.contents)
-                            representationContent.hash = representationContent.url[(representationContent.url.LastIndexOf('/') + 1)..];
+                        data = new WearableItem.Data
+                        {
+                            representations = new WearableItem.Representation[metadata.data.representations.Length],
+                            category = metadata.data.category,
+                            hides = metadata.data.hides,
+                            replaces = metadata.data.replaces,
+                            tags = metadata.data.tags,
+                        },
+                        baseUrl = $"{catalyst.contentUrl}/contents/",
+                        baseUrlBundles = ASSET_BUNDLES_URL_ORG,
+                        emoteDataV0 = null,
+                        description = metadata.description,
+                        i18n = metadata.i18n,
+                        id = metadata.id,
+                        rarity = metadata.rarity,
+                        thumbnail = entity.GetContentHashByFileName(metadata.thumbnail),
+                        MostRecentTransferredDate = DateTimeOffset.FromUnixTimeSeconds(item.GetMostRecentTransferTimestamp())
+                                                                  .DateTime,
+                    };
+
+                    for (var i = 0; i < metadata.data.representations.Length; i++)
+                    {
+                        WearableWithEntityResponseDto.ElementDto.EntityDto.MetadataDto.Representation representation = metadata.data.representations[i];
+
+                        wearable.data.representations[i] = new WearableItem.Representation
+                        {
+                            bodyShapes = representation.bodyShapes,
+                            mainFile = representation.mainFile,
+                            overrideHides = representation.overrideHides,
+                            overrideReplaces = representation.overrideReplaces,
+                            contents = new WearableItem.MappingPair[representation.contents.Length],
+                        };
+
+                        for (var z = 0; z < representation.contents.Length; z++)
+                        {
+                            string fileName = representation.contents[z];
+                            string hash = entity.GetContentHashByFileName(fileName);
+
+                            wearable.data.representations[i].contents[z] = new WearableItem.MappingPair
+                            {
+                                url = $"{catalyst.contentUrl}/contents/{hash}",
+                                hash = hash,
+                                key = fileName,
+                            };
+                        }
```

Comment:
Let's move all this to a static function inside `WearableItem` or a Utils library. Something like `From WearableMetadataDto`.